### PR TITLE
Display "Too few inputs for table" if Karnaugh Map has only 1 input

### LINKED ISSUE #578: Combinational Analysis: IllegalArgumentException if only one input and one output are defined
If one adds only one input variable and one output variable in the `Combinational Analysis` window then clicking on `Minimized` will throw a `java.lang.IllegalArgumentException: Can't add attribute to 0-length text`:

image

This is the full stack trace:

```
Exception in thread "AWT-EventQueue-0" java.lang.IllegalArgumentException: Can't add attribute to 0-length text
        at java.base/java.text.AttributedString.addAttribute(AttributedString.java:322)
        at com.cburch.logisim.analyze.gui.KarnaughMapPanel.Styled(KarnaughMapPanel.java:667)
        at com.cburch.logisim.analyze.gui.KarnaughMapPanel.Styled(KarnaughMapPanel.java:676)
        at com.cburch.logisim.analyze.gui.KarnaughMapPanel.drawNumberedHeader(KarnaughMapPanel.java:623)
        at com.cburch.logisim.analyze.gui.KarnaughMapPanel.paintKmap(KarnaughMapPanel.java:522)
        at com.cburch.logisim.analyze.gui.KarnaughMapPanel.paintComponent(KarnaughMapPanel.java:476)
        at java.desktop/javax.swing.JComponent.paint(JComponent.java:1074)
        at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
        at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
        at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
        at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
        at java.desktop/javax.swing.JViewport.paint(JViewport.java:737)
        at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
        at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
        at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
        at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
        at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:907)
        at java.desktop/javax.swing.JComponent.paint(JComponent.java:1083)
        at java.desktop/javax.swing.JComponent.paintToOffscreen(JComponent.java:5264)
        at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBufferedImpl(RepaintManager.java:1643)
        at java.desktop/javax.swing.RepaintManager$PaintManager.paintDoubleBuffered(RepaintManager.java:1618)
        at java.desktop/javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1556)
        at java.desktop/javax.swing.RepaintManager.paint(RepaintManager.java:1323)
        at java.desktop/javax.swing.JComponent._paintImmediately(JComponent.java:5212)
        at java.desktop/javax.swing.JComponent.paintImmediately(JComponent.java:5022)
        at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:865)
        at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:848)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
        at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:848)
        at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:823)
        at java.desktop/javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:772)
        at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1884)
        at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:316)
        at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
        at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
        at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

The error occurs on the current HEAD of branch develop, i.e., commit 32152c1524d81c2d5a62142eee92fa9d27c24533.

### PR DESCRIPTION
**Pull request by simoc**
If there is only one input to a Karnaugh Map, then display the message "Too few inputs for table" instead.

A Karnaugh Map only makes sense if there are at least two inputs.

I have added this message string to the `analyze.properties` files for some languages, but left the placeholder `# ==> karnaughTooFewInputsError =` for the languages for which I have no idea of the translation.

Fixes #578

### CODE REVIEW

File: src/main/resources/resources/logisim/strings/analyze/analyze_ja.properties [hunk 1]
```diff
@@ -120,6 +120,7 @@ openErrorTitle = ファイルの読み取りエラー
 #
 karnaughNoInputsError = 利用可能な入力がありません。
 karnaughNoOutputError = 選択された出力がありません。
+# ==> karnaughTooFewInputsError =
```
Comment (maehne) [inline src/main/resources/resources/logisim/strings/analyze/analyze_ja.properties:123]: @Narita-1095305: Could you maybe provide a Japanese translation for the new "Too few inputs for table." message? DeepL has several [suggestions](https://www.deepl.com/translator#en/ja/Too%20few%20inputs%20for%20the%20table.), but Japanese is just too far away from the languages I know.

File: src/main/resources/resources/logisim/strings/analyze/analyze_pt.properties [hunk 2]
```diff
@@ -120,6 +120,7 @@ openErrorTitle = Arquivo de leitura de erros
 #
 karnaughNoInputsError = Não há entradas disponíveis.
 karnaughNoOutputError = Nenhuma saída selecionada.
+# ==> karnaughTooFewInputsError =
```
Comment (maehne) [inline src/main/resources/resources/logisim/strings/analyze/analyze_pt.properties:123]: [DeepL](https://www.deepl.com/translator) suggests: "Entradas demasiado poucas para a tabela."

File: src/main/resources/resources/logisim/strings/analyze/analyze_ru.properties [hunk 3]
```diff
@@ -120,6 +120,7 @@ openErrorTitle = Файл считывания ошибок
 #
 karnaughNoInputsError = Входы недоступны.
 karnaughNoOutputError = Выход не выбран.
+# ==> karnaughTooFewInputsError =
```
Comment (maehne) [inline src/main/resources/resources/logisim/strings/analyze/analyze_ru.properties:123]: [DeepL](https://www.deepl.com/translator#en/ru/Too%20few%20inputs%20for%20the%20table.) suggests "Слишком мало входов для таблицы."

File: src/main/resources/resources/logisim/strings/analyze/analyze_el.properties [hunk 4]
```diff
@@ -120,6 +120,7 @@ UsedSeperatorInFile = Επιλέξτε το χαρακτήρα διαχωρισ
 #
 karnaughNoInputsError = Δεν υπάρχουν διαθέσιμες εισροές.
 karnaughNoOutputError = Καμία επιλεγμένη έξοδο.
+# ==> karnaughTooFewInputsError =
```
Comment (maehne) [inline src/main/resources/resources/logisim/strings/analyze/analyze_el.properties:123]: [DeepL](https://www.deepl.com/translator#en/el/Too%20few%20inputs%20for%20the%20table.) suggests "Πολύ λίγες είσοδοι για τον πίνακα."

**Review by maehne — APPROVED** [review]
Thanks @simoc for proposing a fix for issue #578! It looks good to me. I added some suggestions for additional localisation strings translated with the help of [DeepL](https://www.deepl.com/). They seem close enough to the already translated "Too many inputs for table." and check out also when translating them back.

Actually, a 1 input Karnaugh map is defined: It would be just two squares next to each other in a row or column labeled with the variable values 0 and 1. There would be just no label left of the row or above the column.

@BFH-ktt1: What do you think? Should we support the 1 input case? It would be consistent how we explain the construction of Karnaugh maps in class.

File: src/main/resources/resources/logisim/strings/analyze/analyze_pt.properties (hunk 2 above)
Comment (simoc) [reply src/main/resources/resources/logisim/strings/analyze/analyze_pt.properties]: Added in commit fa06451

File: src/main/resources/resources/logisim/strings/analyze/analyze_ru.properties (hunk 3 above)
Comment (simoc) [reply src/main/resources/resources/logisim/strings/analyze/analyze_ru.properties]: Added in commit fa06451

File: src/main/resources/resources/logisim/strings/analyze/analyze_el.properties (hunk 4 above)
Comment (simoc) [reply src/main/resources/resources/logisim/strings/analyze/analyze_el.properties]: Added in commit fa06451

**Pull request merged**